api/src/modules/tasks/dto/update-task-with-controller.dto.ts

```diff
@@ -0,0 +1,23 @@
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { IsNotEmpty, IsOptional, IsString, IsUUID } from 'class-validator';
+import { TASK_STATUS } from 'modules/tasks/task.entity';
+
+export class UpdateTaskWithControllerDto {
+  @ApiProperty()
+  @IsNotEmpty()
+  @IsUUID()
+  taskId: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  @IsString()
+  newStatus: TASK_STATUS;
```

Comment:
```suggestion
  newStatus?: TASK_STATUS;
```

Reply:
overall I think you need to account for updates on things other than status (even if those are a bit strange)

Reply:
I agree. I believe here the newStatus is required since the method UPDATING a status **within the api** requires a new status, but if we want to let a user/admin update a task, we should allow to update another fields as well

Reply:
so now, when updating Task, taskId is required, rest is optional for user to update (change status, add data, add errors). This endpoint doesn't allow to add logs, since it is not part of the updateImportJobEvent service used in the endpoint, I can add it to service or @alexeh, what do you think, probably I should create separate method for updating Task via controller to to mix it with the updates coming from import process?

Reply:
I would say don't worry too much about this endpoint, as it's more of a maintenance tool than anything else. Based on my experience, the most useful part will be updating status - in case a job gets stuck importing, and we need to fix it to unblock newer imports.

Reply:
okay, so left possibility to update status and data via controller. If at some point we will need to be able to remove an error or logs, probably there should be a different, a bit more complex endpoint, since those columns are arrays of objects now..